service/block/event_test.go

```diff
@@ -73,16 +122,43 @@ func (m *mockFetcher) UnsubscribeNewBlockEvent(ctx context.Context) error {
 	return nil
 }
 
+func (m *mockFetcher) generateBlocksWithValidHeaders(t *testing.T, num int) []*RawBlock {
+	suite := header.NewTestSuite(t, 3)
+	rawBlocks := make([]*RawBlock, num)
+
+	prevEH := suite.GenExtendedHeader()
+
+	for i := range rawBlocks {
+		eh := suite.GenExtendedHeader()
+		b := &RawBlock{
+			Header:     eh.RawHeader,
+			LastCommit: prevEH.Commit,
+		}
+		require.NoError(t, b.ValidateBasic())
+		require.NoError(t, b.Header.ValidateBasic())
+
+		rawBlocks[i] = b
+		// store commit and valset at height
+		m.commits[b.Height] = eh.Commit
+		m.valSets[b.Height] = eh.ValidatorSet
+
+		m.mockNewBlockCh <- b
+		prevEH = eh
+	}
+	return rawBlocks
+}
+
 // generateBlocks generates new raw blocks and sends them to the mock fetcher,
 // returning the extended blocks generated from the process to compare against.
 func (m *mockFetcher) generateBlocks(t *testing.T, num int) []Block {
 	t.Helper()
 
 	extendedBlocks := make([]Block, num)
 
-	for i := 0; i < num; i++ {
+	for i := 1; i <= num; i++ {
```

Comment:
Why this change? Doesn't seem to have any effects?

Reply:
relic from when I was debugging something - you're right